### PR TITLE
Adding Spacing Between and other fun stuff

### PR DESCRIPTION
**Pull request by rbnswartz**

### CODE REVIEW

File: ooxml/XWPF/Usermodel/XWPFDocument.cs [hunk 1]
```diff
@@ -291,8 +291,21 @@ public CT_Document Document
...
             }
+            set
+            {
+                ctDocument = value;
+            }
+        }
+        /**
+         * Sets columns on document base object
+         */
+        public void setColumnCount(int num)
```
Comment (rbnswartz) [inline ooxml/XWPF/Usermodel/XWPFDocument.cs]: Make this a property of ColumnCount

File: ooxml/XWPF/Usermodel/XWPFParagraph.cs [hunk 2]
```diff
@@ -958,11 +958,14 @@ public int SpacingBeforeLines
             }
         }
 
-        /// <summary>
-        ///Specifies how the spacing between lines is calculated as stored in the
-        /// line attribute. If this attribute is omitted, then it shall be assumed to
-        /// be of a value auto if a line attribute value is present.
-        /// </summary>
+        /**
```
Comment (rbnswartz) [inline ooxml/XWPF/Usermodel/XWPFParagraph.cs]: Fix this javadoc

File: ooxml/XWPF/Usermodel/XWPFParagraph.cs [hunk 3]
```diff
@@ -977,7 +980,42 @@ public LineSpacingRule SpacingLineRule
                 spacing.lineRule = EnumConverter.ValueOf<ST_LineSpacingRule, LineSpacingRule>(value);
             }
         }
+        /**
+        * Return the spacing between lines of a paragraph. The units of the return value depends on the
```
Comment (rbnswartz) [inline ooxml/XWPF/Usermodel/XWPFParagraph.cs]: This one too.

**Review by rbnswartz — COMMENTED** [review]
Some things need to be changed

File: ooxml/XWPF/Usermodel/XWPFParagraph.cs (hunk 2 above)
Comment (BryanHo10) [reply ooxml/XWPF/Usermodel/XWPFParagraph.cs]: Changed to C# doc

File: ooxml/XWPF/Usermodel/XWPFDocument.cs (hunk 1 above)
Comment (BryanHo10) [reply ooxml/XWPF/Usermodel/XWPFDocument.cs]: Created property columnCountField

File: ooxml/XWPF/Usermodel/XWPFDocument.cs [hunk 4]
```diff
@@ -299,12 +300,20 @@ public CT_Document Document
...
          * Sets columns on document base object
          */
-        public void setColumnCount(int num)
+        public int ColumnCount
         {
-            if(ctDocument != null)
+            get
             {
-                ctDocument.body.sectPr.cols.num = num.ToString();
+                return this.columnCountField;
```
Comment (rbnswartz) [inline ooxml/XWPF/Usermodel/XWPFDocument.cs]: I think this is an actual error nothing is ever going to come out of this (that I can see) it should be taking and parsing the value back out.

File: ooxml/XWPF/Usermodel/XWPFDocument.cs [hunk 5]
```diff
@@ -291,8 +292,39 @@ public CT_Document Document
...
+                {
+                    ctDocument.body.sectPr.cols.num = value.ToString();
+                }
+            }
+            
+        }
+        /**
+         * Sets Text Direction of Document
+         */
+         public void setTextDirection(ST_TextDirection direction)
```
Comment (rbnswartz) [inline ooxml/XWPF/Usermodel/XWPFDocument.cs]: This should probably be a property also

File: ooxml/XWPF/Usermodel/XWPFParagraph.cs [hunk 6]
```diff
@@ -978,6 +980,36 @@ public LineSpacingRule SpacingLineRule
             }
         }
 
+        ///<summary>
```
Comment (rbnswartz) [inline ooxml/XWPF/Usermodel/XWPFParagraph.cs:983]: Thank you!

File: ooxml/XWPF/Usermodel/XWPFDocument.cs (hunk 4 above)
Comment (BryanHo10) [reply ooxml/XWPF/Usermodel/XWPFDocument.cs]: Yeah you're right. I'll return the parsed value  from [ctDocument.body.sectPr.cols.num]

File: ooxml/XWPF/Usermodel/XWPFDocument.cs (hunk 5 above)
Comment (BryanHo10) [reply ooxml/XWPF/Usermodel/XWPFDocument.cs]: Yup I'll mirror the ColumnCount

**Review by rbnswartz — COMMENTED** [review]
Looks good lets merge it in! @BryanHo10

**Comment by BryanHo10** [comment]
@rbnswartz  It's weird, github says I don't have write access to this repository. I think you need to add me as collaborator

**Pull request merged**